catalog/ui/catalog-ui-search/src/main/webapp/js/controllers/cesium.layerCollection.controller.js

```diff
@@ -60,6 +60,20 @@ define(['underscore',
                 }
 
                 var provider = new type(initObj);
+
+                /*
+                  Optionally add this provider as a TrustedServer. This sets withCredentials = true
+                  on the XmlHttpRequests for CORS.
+                 */
+                if (model.get('withCredentials')) {
+                  var parsedUrl = new URL(provider.url);
```

Comment:
❗ `URL` is not standard and will not work in Internet Explorer.  

http://caniuse.com/#search=URL

Reply:
Any suggestions on what to use instead?

Reply:
Cesium uses (and should include maybe) this: https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/ThirdParty/Uri.js

I wonder if we can just piggyback that, since it already needs to work for the TrustedServer stuff to work anyway.

Note in both that logic, and in ours, scheme gets parsed as "https:" or "http:", so we will either need to use a startsWith or a length-1 for our comparison in line 72

Reply:
I think you can use `const url = require('url')` instead. Then do something like `url.parse(provider.url)` which returns an object with port, protocol, and hostname.